concepts/pointers/about.md

```diff
@@ -1,3 +1,154 @@
+
+```go
+var a int 
+a = 2
+
+var p *int 
+p = &a // the variable 'p' will hold the memory address of 'a'
+
+var b int
+b = *p // b == 2
+```
+
+The operation `*p` fetches the value stored at the memory address stored in `p`. This operation is often called "dereferencing".
+
+We can also use the derefering operator to assign a new value to the memory address referenced by the pointer:
+
+```go
+var a int
+a = 2   // declare int variable 'a' and assign it the value of 2
+
+var pa *int
+pa = &a          // 'pa' now refers to the memory address of 'a'  
+*pa = *pa + 2    // increment by 2 the value at memory address 'pa'
+
+fmt.Println(a)   // Output: 4
+                 // 'a' will have the new value that was changed via the pointer!
+```
+
+Assigning to `*pa` will change the value stored at the memory address `pa` holds. Since `pa` holds the memory address of `a`, by assigning to `*pa` we are effectively changing the value of `a`! 
+
+A note of caution however: always check if a pointer is not `nil` before dereferencing. Dereferencing a `nil` pointer will make the program crash at runtime!
+
+### Pointers to structs
+
+So far we've only seen pointers to primitive values. We can also create pointers for structs:
+
+```go
+type Person struct {
+    Name string
+    Age  int
+}
+
+var peter Person
+peter = Person{Name: "Peter", Age: 22}
+
+var p *Person
+p = &peter
+```
+
+We could have also created a new `Person` and immediately stored a pointer to it:
+
+```go
+var p *Person
+p = &Person{Name: "Peter", Age: 22}
+```
+
+When we have a pointer to a struct, we don't need to dereference the pointer before accessing one of the fields:
+
+```go
+var p *Person
+p = &Person{Name: "Peter", Age: 22}
+
+fmt.Println(p.Name) // Output: "Peter" 
+                    // Go automatically dereferences 'person' to allow
+                    // access to the 'Name' field
+```
+
+## Slices and maps are already pointers
+
+Slices and maps are special types because they already have pointers in their implementation. This means that more often that not, we don't need to create pointers for these types to share the memory address for their values. Imagine we have a function that adds a key to a map:
+
+
+```go
+func addPeterAge(m map[string]int){
+	m["Peter"] = 22
+}
+```
+
+If we create a map and call this function, the changes the function made to the map persist after the function ended. This is a similar behavior we get if we were using a pointer, but note how on this example we are not using any referencing/dereferencing or any of the pointer syntax:
+
+```go
+ages := make(map[string]int)
+addPeterAge(ages)
+fmt.Println(ages)
+// Output: map[Peter:22]
+// The changes the function 'addPeterAge' made to the map are visible after the function ends!
+```
+
+## Pointer arithmetic
+
+Unlike other languages, pointer arithmetic is not allowed in Go. This means that snippets like the following are invalid:
```

Comment:
Suggesting to add an explanation of that pointer arithmetic actually is. Also rephrased the first bit a little.
```suggestion
Unlike other languages, Go does not support pointer arithmetic. This means you cannot go from a memory address of one value to the address of another by incrementing the pointer value. So snippets like the following are invalid:
```